qiskit_nature/second_q/transformers/base_transformer.py

```diff
@@ -25,13 +26,32 @@ class BaseTransformer(ABC):
 
     @abstractmethod
     def transform(self, problem: BaseProblem) -> BaseProblem:
-        """Transforms one :class:`~qiskit_nature.second_q.problems.BaseProblem` into another one.
+        """Transforms one :class:`~qiskit_nature.second_q.problems.BaseProblem` into another.
         This may or may not affect the size of the Hilbert space.
 
         Args:
             problem: the problem to be transformed.
 
+        Raises:
+            NotImplementedError: when an unsupported problem type is provided.
+
         Returns:
             A new `BaseProblem` instance.
         """
         raise NotImplementedError()
+
+    @abstractmethod
+    def transform_hamiltonian(self, hamiltonian: Hamiltonian) -> Hamiltonian:
+        """Transforms one :class:`~qiskit_nature.second_q.hamiltonians.Hamiltonian` into another.
+        This may or may not affect the size of the Hilbert space.
```

Comment:
Very minor comment, like the above transform `This may or may not affect the size of the Hilbert space.` I think `This may affect the size of the Hilbert space.` is entirely equivalent right - the `or may not` is implicit in `may affect`